File: uikit/src/main/java/listeners/CustomAlertDialogHelper.java
package listeners;

import android.content.Context;
import android.content.DialogInterface;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AlertDialog;

import com.cometchat.pro.uikit.R;

public class CustomAlertDialogHelper implements View.OnClickListener {
    private static final String TAG = CustomAlertDialogHelper.class.getSimpleName();

    private OnAlertDialogButtonClickListener onAlertDialogButtonClick;

    private View view;

    private AlertDialog alertDialogCreater;

    private int popupId;
    private int colorPrimary;

    //	cc cometChat;
    public CustomAlertDialogHelper(Context context, String title, View view, String positiveTitle, String neutralTitle,
                                   String negativeTitle, OnAlertDialogButtonClickListener onAlertDialogButton, int popUpId, boolean isCancelable) {
        onAlertDialogButtonClick = onAlertDialogButton;
        // LayoutInflater inflater = (LayoutInflater)
        // context_menu.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        this.view = view;
        AlertDialog.Builder builder = new AlertDialog.Builder(context);

        //todo get color from cc Sdk
//		cometChat = cc.getInstance(context_menu);
        colorPrimary = context.getResources().getColor(R.color.colorPrimaryDark);
//		colorPrimary = (int) cometChat.getCCSetting(new CCSettingMapper(SettingType.UI_SETTINGS, SettingSubType.COLOR_PRIMARY));
        builder.setView(view);
        builder.setCancelable(isCancelable);
        if (!title.equals("")) {
            builder.setTitle(title);
        }


        if (!positiveTitle.equals("")) {
            builder.setPositiveButton(positiveTitle, null);
        }
        if (!negativeTitle.equals("")) {
            builder.setNegativeButton(negativeTitle, null);
        }
        if (!neutralTitle.equals("")) {
            builder.setNeutralButton(neutralTitle, null);
        }

        alertDialogCreater = builder.create();
        alertDialogCreater.show();

        this.popupId = popUpId;

        Button positiveButton = alertDialogCreater.getButton(DialogInterface.BUTTON_POSITIVE);
        positiveButton.setId(DialogInterface.BUTTON_POSITIVE);
        positiveButton.setTextColor(colorPrimary);
        positiveButton.setOnClickListener(this);

        Button negativeButton = alertDialogCreater.getButton(DialogInterface.BUTTON_NEGATIVE);
        negativeButton.setId(DialogInterface.BUTTON_NEGATIVE);
        negativeButton.setTextColor(colorPrimary);
        negativeButton.setOnClickListener(this);

        Button neutralButton = alertDialogCreater.getButton(DialogInterface.BUTTON_NEUTRAL);
        neutralButton.setId(DialogInterface.BUTTON_NEUTRAL);
        neutralButton.setTextColor(colorPrimary);
        neutralButton.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        onAlertDialogButtonClick.onButtonClick(alertDialogCreater, view, v.getId(), popupId);
    }

}

File: uikit-kotlin/src/main/java/com/cometchat/pro/uikit/ui_resources/utils/custom_alertDialog/CustomAlertDialogHelper.kt
package com.cometchat.pro.uikit.ui_resources.utils.custom_alertDialog

import android.content.Context
import android.content.DialogInterface
import android.view.View
import androidx.appcompat.app.AlertDialog
import com.cometchat.pro.uikit.R

class CustomAlertDialogHelper(context: Context, title: String, private val view: View, positiveTitle: String, neutralTitle: String,
                              negativeTitle: String, private val onAlertDialogButtonClick: OnAlertDialogButtonClickListener, popUpId: Int, isCancelable: Boolean) : View.OnClickListener {
    private val alertDialogCreater: AlertDialog
    private val popupId: Int
    private val colorPrimary: Int
    override fun onClick(v: View) {
        onAlertDialogButtonClick.onButtonClick(alertDialogCreater, view, v.id, popupId)
    }

    companion object {
        private val TAG = CustomAlertDialogHelper::class.java.simpleName
    }

    //	cc cometChat;
    init {
        // LayoutInflater inflater = (LayoutInflater)
        // context_menu.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        val builder = AlertDialog.Builder(context)

        //todo get color from cc Sdk
//		cometChat = cc.getInstance(context_menu);
        colorPrimary = context.resources.getColor(R.color.colorPrimaryDark)
        //		colorPrimary = (int) cometChat.getCCSetting(new CCSettingMapper(SettingType.UI_SETTINGS, SettingSubType.COLOR_PRIMARY));
        builder.setView(view)
        builder.setCancelable(isCancelable)
        if (title != "") {
            builder.setTitle(title)
        }
        if (positiveTitle != "") {
            builder.setPositiveButton(positiveTitle, null)
        }
        if (negativeTitle != "") {
            builder.setNegativeButton(negativeTitle, null)
        }
        if (neutralTitle != "") {
            builder.setNeutralButton(neutralTitle, null)
        }
        alertDialogCreater = builder.create()
        alertDialogCreater.show()
        popupId = popUpId
        val positiveButton = alertDialogCreater.getButton(DialogInterface.BUTTON_POSITIVE)
        positiveButton.id = DialogInterface.BUTTON_POSITIVE
        positiveButton.setTextColor(colorPrimary)
        positiveButton.setOnClickListener(this)
        val negativeButton = alertDialogCreater.getButton(DialogInterface.BUTTON_NEGATIVE)
        negativeButton.id = DialogInterface.BUTTON_NEGATIVE
        negativeButton.setTextColor(colorPrimary)
        negativeButton.setOnClickListener(this)
        val neutralButton = alertDialogCreater.getButton(DialogInterface.BUTTON_NEUTRAL)
        neutralButton.id = DialogInterface.BUTTON_NEUTRAL
        neutralButton.setTextColor(colorPrimary)
        neutralButton.setOnClickListener(this)
    }
}